src/hooks/useLocation.js

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
+import RNLocation from 'react-native-location';
+
+import { INITIAL_COORDS, DISTANCE_FILTER } from 'constants/common';
+import parseError from 'utils/parseError';
+
+RNLocation.configure({
+  distanceFilter: DISTANCE_FILTER,
+});
+
+const useLocation = () => {
+  const [location, setLocation] = useState(INITIAL_COORDS);
+
+  useEffect(() => {
+    async function requestLocation() {
+      try {
+        let granted = await RNLocation.checkPermission({
+          ios: 'whenInUse',
+          android: {
+            detail: 'coarse',
+          },
+        });
+        if (!granted) {
+          granted = await RNLocation.requestPermission({
+            ios: 'whenInUse',
+            android: {
+              detail: 'coarse',
+            },
+          });
+        }
+        if (granted) {
+          const latestLocation = await RNLocation.getLatestLocation({ timeout: 60000 });
+          const { latitude, longitude } = latestLocation;
+          setLocation({ latitude, longitude });
```

Comment:
You can destructure in the first line, don't need to do it separately. 
`const { latitude, longitude } = await RNLocation.getLatestLocation({ timeout: 60000 });`